.github/workflows/code/test-migrations/index.js

```diff
@@ -0,0 +1,158 @@
+require('bluebird-global')
+const exec = require('child_process').exec
+const archive = require('../../../../out/bp/core/misc/archive')
+const fs = require('fs')
+const rimraf = require('rimraf')
+const path = require('path')
+const glob = require('glob')
+const semver = require('semver')
+const _ = require('lodash')
+const AWS = require('aws-sdk')
+const chalk = require('chalk')
+const core = require('@actions/core')
+const github = require('@actions/github')
+
+const ensureDownMigration = async () => {
+  const pullRequest = JSON.parse(process.env.PULL_REQUEST)
+  const octokit = github.getOctokit(process.env.TOKEN)
+  const options = {
+    repo: 'botpress',
+    owner: 'botpress',
+    pull_number: pullRequest.number,
+    per_page: 300
+  }
+
+  try {
+    const { data: files } = await octokit.pulls.listFiles(options)
+
+    for (const { filename } of files.filter(x => x.filename.includes('/migrations/'))) {
+      const fileContent = fs.readFileSync(filename, 'UTF-8').toString()
+
+      if (!fileContent.includes('down: ')) {
+        core.setFailed('Migrations must also implement the "down" method to revert back to a previous version.')
+      }
+    }
+  } catch (err) {
+    core.setFailed(err.message)
+  }
+}
+
+const start = async () => {
+  await ensureDownMigration()
+
+  const targetVersion = getMostRecentVersion()
+
+  const Bucket = 'botpress-migrations'
+  const s3 = new AWS.S3({
+    accessKeyId: process.env.AWS_ACCESS_KEY_ID,
+    secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY
+  })
+
+  const dir = await Promise.fromCallback(cb => s3.listObjectsV2({ Bucket }, cb))
+  for (const file of dir.Contents) {
```

Comment:
you test migrations for all dumps ?  like forever ?

Reply:
For the moment, yup. The mechanism is there, now we can discuss what type of migration we want to test